sdk-server expects SDK_LOG_LEVEL

KAllan357 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/googleforgames/agones/blob/main/CONTRIBUTING.md and developer guide https://github.com/googleforgames/agones/blob/main/build/README.md
2. Please label this pull request according to what type of issue you are addressing.
3. Ensure you have added or ran the appropriate tests for your PR: https://github.com/googleforgames/agones/blob/main/build/README.md#testing-and-building
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:
I think I noticed a bug in my Agones sdk-server where I was constantly seeing the log line:

```
{"error":"not a valid logrus Level: \"\"","message":"Invalid LOG_LEVEL value. Defaulting to 'info'.","severity":"warning","time":"2024-06-05T13:53:37.5359166-04:00"}
```

I'm also seeing this in my local K8s cluster running Agones v1.40:
```
agones-gameserver-sidecar {"error":"not a valid logrus Level: \"\"","message":"Invalid LOG_LEVEL value. Defaulting to 'info'.","severity":"warning","time":"2024-06-05T18:08:05.006261795Z"}
```

There is a chance this is a bug introduced in #3631. The relevant code in the sdk-server binary is

https://github.com/googleforgames/agones/blob/main/cmd/sdk-server/main.go#L326
```
runtime.Must(viper.BindEnv("logLevel", "SDK_LOG_LEVEL"))
```

This PR simply adjusts the name Env Var attached to the Gameserver's Agones SDK Sidecar container.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Closes #<issue number>`, or `Closes (paste link of issue)`.
-->
Closes #

**Special notes for your reviewer**:

markmandel reviewed (commented):
Ooh! Good catch. I didn't see that in the PR!

Rather than change Env var settings from what they used to be (should be?), I'd suggest fixing:

https://github.com/googleforgames/agones/blob/bc42fd5c195d7c29e1cd2a0776f3f448d1d0dda0/cmd/sdk-server/main.go#L326

And setup a proper `logLevelFlag` variable etc, and make the above line be:

```go
runtime.Must(viper.BindEnv(logLevelFlag))
```

Then we fix the bug in the sdk-server rather than potentially force an env var change.

Also that default here: https://github.com/googleforgames/agones/blob/bc42fd5c195d7c29e1cd2a0776f3f448d1d0dda0/cmd/sdk-server/main.go#L290 -- that's no good. It should also be "info" 😄

KAllan357 commented:
@markmandel Thanks for the feedback. I adjusted the PR as you described. Note that I also changed it from `"logLevel"` to `log-level` to get Viper to do its thing when the env var is `LOG_LEVEL`:

```
$ go run cmd/sdk-server/main.go
{"ctlConf":{"GameServerName":"","PodNamespace":"".....

$ LOG_LEVEL=HOWDY go run cmd/sdk-server/main.go
{"error":"not a valid logrus Level: \"HOWDY\"","message":"Invalid LOG_LEVEL value. Defaulting to 'info'.","severity":"warning","time":"2024-06-06T13:22:19.982728509-04:00"}
{"ctlConf":{"GameServerName":"","PodNamespace":"" .....
```

markmandel commented:
Thanks! Sounds good - will take a peek next week 👍🏻

markmandel reviewed (approved):
Looks great, thanks!

Shavakan commented:
We also faced a similar issue recently and ended up in this PR.
@markmandel I hope I don't sound pushing, but could you kindly inform me when this fix will likely be released?
